src/components/leaf.js

```diff
@@ -148,17 +163,47 @@ class Leaf extends React.Component {
     const hasFocus = selection.hasFocusBetween(node, start, end)
     if (!hasAnchor && !hasFocus) return
 
+    // We have a selection to render, so prepare a few things...
+    const ref = ReactDOM.findDOMNode(this)
+    const el = findDeepestNode(ref)
+    const window = getWindow(el)
+
+    // If no selection is defined we're probably in a node environment and we can skip the selection magic.
+    if (!window || !window.getSelection) return
+
+    const native = window.getSelection()
+    const parent = ref.closest('[contenteditable]')
+
+    // If the selection is blurred but the DOM selection still focuses this leaf,
+    // we need to clean up the ranges and blur the contenteditable.
+    if (selection.isBlurred && (hasAnchor || hasFocus)) {
+      // We need to make sure that the selection from our state is up-to-date with the native selection.
+      // We can do this by checking if native.anchorNode is a descendant of our contenteditable parent, which is this
+      // slate instance. If it's not, the selection is no longer under this instance's responsibility and not further
+      // action is required.
+      if (!isDescendant(parent, native.anchorNode)) return
+
+      // Apparently our selection is blurred, but the DOM still has ranges in the nodes we manage. To fix this,
+      // we blur the contenteditable and remove all ranges.
+      native.removeAllRanges()
+      if (parent) parent.blur()
```

Comment:
No, that's exactly the point of this PR, it's unfortunately not enough to just removeAllRanges, you also need to blur the contenteditable or otherwise key strokes will be processed by the onInput event in all browsers I tested